src/Sylius/Component/Resource/Repository/InMemoryRepository.php

```diff
@@ -0,0 +1,242 @@
+     */
+    public function findAll()
+    {
+        return $this->arrayObject->getArrayCopy();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+    {
+        $results = $this->findAll();
+
+        if (!empty($criteria)) {
+            $results = $this->applyCriteria($results, $criteria);
+        }
+
+        if (isset($orderBy)) {
+            $results = $this->applyOrder($results, $orderBy);
+        }
+
+        $results = array_slice($results, $offset, $limit);
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function findOneBy(array $criteria)
+    {
+        if (empty($criteria)){
+            throw new \InvalidArgumentException('The criteria array needs to be set.');
+        }
+
+        $results = $this->applyCriteria($this->findAll(), $criteria);
+
+        if ($result = reset($results)) {
+            return $result;
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getClassName()
+    {
+        return $this->interface;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createPaginator(array $criteria = null, array $orderBy = null)
+    {
+        $results = $this->findAll();
+
+        if (!empty($orderBy)) {
+            $results = $this->applyOrder($results, $orderBy);
+        }
+
+        if (!empty($criteria)) {
+            $results = $this->applyCriteria($results, $criteria);
+        }
+
+        $adapter = new ArrayAdapter($results);
+        $pagerfanta = new Pagerfanta($adapter);
+
+        return $pagerfanta;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNew()
+    {
+    }
+
+    /**
+     * @param ResourceInterface[] $resources
+     * @param array               $criteria
+     *
+     * @return ResourceInterface[]|array
+     */
+    private function applyCriteria(array $resources, array $criteria)
+    {
+        foreach ($this->arrayObject as $object) {
+            foreach ($criteria as $criterion => $value) {
+                if ($value !== $this->accessor->getValue($object, $criterion)) {
+                    $key = array_search($object, $resources);
+                    unset($resources[$key]);
+                }
+            }
+        }
+
+        return $resources;
+    }
+
+    /**
+     * @param ResourceInterface[] $resources
+     * @param array               $orderBy
+     *
+     * @return ResourceInterface[]
+     */
+    private function applyOrder(array $resources, array $orderBy)
+    {
+        $property = key($orderBy);
+        $order = $orderBy[$property];
+
+        $sortable = array();
+        $results = array();
+
+        foreach ($resources as $key => $object) {
+            $sortable[$key] = $this->accessor->getValue($object, $property);
+        }
+
+        if ('ASC' === $order) {
```

Comment:
Hm... not sure if it's worth it if the rest of logic stays the same. That being said, that is not related to the usage of a constant here, hm? $order is coming from the user, so I think we should have constants here.